core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4792,6 +4828,35 @@ private void convertValuesImpl(
         true);
   }
 
+  private int[] processColumnsMapping(RelDataType targetRowType, RelOptTable targetTable) {
+    List<ColumnStrategy> strategies = targetTable.getColumnStrategies();
+    List<String> targetFields = targetRowType.getFieldNames();
+    List<RelDataTypeField> tblFields = targetTable.getRowType().getFieldList();
+
+    int[] mapping = new int[targetFields.size()];
+
+    Arrays.fill(mapping, -1);
+
+    Ord.forEach(tblFields, (fld, i) -> {
+      if (strategies.get(i) == ColumnStrategy.DEFAULT) {
+        int pos = targetFields.indexOf(fld.getName());
+        if (pos != -1) {
+          mapping[pos] = i;
+        }
+      }
+    });
+
+    if (this.getClass().desiredAssertionStatus()) {
```

Comment:
I'd remove this `if`. This is not performance-critical code.  The assertion is worth checking even if assertions are disabled.